_posts/2019-01-26-generic-way-of-handling-loading-saving-and-validationerrors-in-angular.md

```diff
@@ -0,0 +1,410 @@
+---
+layout: post
+title: A generic way of handling loading-status, saving-status and validation errors in Angular
+published: false
+description: In this article we are going to implement a generic solution on how to fix 3 common usecases that involve redundancy in CRUD applications.
+
+author: brechtbilliet
+comments: true
+date: 2019-01-26
+subclass: 'post'
+categories: 'brechtbilliet'
+disqus: true
+tags: Angular
+cover: 'assets/images/cover/cover2.jpg'
+---
+
+# A generic way of handling loading-status, saving-status and validation errors in Angular
+
+When writing Angular applications there are always pieces of functionality that are being rewritten over and over again. 3 common usecases are:
+
+- Showing a *loading* status
+- Showing an *acting* status (whether the user is adding, updating or removing data)
+- Showing validation errors
+
+In this article we are going to implement a generic solution on how to fix these usecases.
+
+## A non generic way of loading, saving and handling validation errors in Angular.
+
+Before we jump to the solution, let's have a look at the impact of a non generic way of handling the previous called functionalities.
+A solution that is often used might look like the following: (keep in mind that this functionality has to be implemented over and over again for every component) 
+
+```typescript
+loading: boolean;
+acting: boolean;
+
+ngOnInit(): void {
+    this.loading = true;
+    const usersCompleted = false;
+    const citiesCompleted = false;
+    this.userService.fetch().subscribe(users => {
```

Comment:
I can say that this shouldn't be completed to disable loading - in case of request error loading might remain true.

Two possible ways to solve:
1. try - catch - finally
2. pipe with catchError